### PR TITLE
Improvements around test_pidfile_contents

### PR DESCRIPTION
**Pull request by exarkun**
https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3983
https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3984
https://tahoe-lafs.org/trac/tahoe-lafs/ticket/4058

### CODE REVIEW

**Comment by exarkun** [comment]
This _still_ fails, at least because `str.split` and the `\w` regexp disagree about whether `\u2000` is whitespace or not.

File: src/allmydata/test/cli/test_run.py [hunk 1]
```diff
@@ -264,17 +265,49 @@ def test_non_numeric_pid(self):
         self.assertThat(runs, Equals([]))
         self.assertThat(result_code, Equals(1))
 
-    good_file_content_re = re.compile(r"\w[0-9]*\w[0-9]*\w")
```
Comment (exarkun) [inline src/allmydata/test/cli/test_run.py:267]: I greatly elaborated this regular expression and removed some subtle inconsistencies with the real parser but note that apart from getting much longer a significant change here is going from `\w` ("word character") to `\s` ("whitespace character") - a pretty dramatic inconsistency, I think.

**Review by meejah — APPROVED** [review]
LGTM, not sure why still draft

**Comment by meejah** [comment]
see https://github.com/tahoe-lafs/tahoe-lafs/pull/1412

**Pull request closed without merging**